apps/sv/src/main/scala/org/lfdecentralizedtrust/splice/sv/automation/delegatebased/ExpireRewardCouponsTrigger.scala

```diff
@@ -43,15 +44,15 @@ class ExpireRewardCouponsTrigger(
       tc: TraceContext
   ): Future[Seq[ExpiredRewardCouponsBatch]] = for {
     dsoRules <- store.getDsoRules()
-    numSvs = dsoRules.payload.svs.size()
     batches <- store
-      .getExpiredRewards(
+      .getExpiredCouponsInBatchesPerRoundAndCouponType(
         dsoRules.domain,
         context.config.enableExpireValidatorFaucet,
+        PageLimit.tryCreate(svTaskContext.delegatelessAutomationExpiredRewardCouponBatchSize),
       )
-      .map(seq =>
-        Random.shuffle(seq).take(svTaskContext.delegatelessAutomationExpiredRewardCouponBatchSize)
```

Comment:
Not sure my thinking is wrong or this logic was wrong.

Reply:
duh yeah looks right